src/twisted/names/dns.py

```diff
@@ -2005,13 +2088,89 @@ class Record_SPF(Record_TXT):
     @ivar data: Freeform text which makes up this record.
 
     @type ttl: L{int}
-    @ivar ttl: The maximum number of seconds which this record should be cached.
+    @ivar ttl: The maximum number of seconds
+               which this record should be cached.
     """
     TYPE = SPF
     fancybasename = 'SPF'
 
 
 
+@implementer(IEncodable, IRecord)
+class Record_TSIG(tputil.FancyEqMixin, tputil.FancyStrMixin):
+    """
+    A transaction signature, encapsulated in a RR, as described
+    in U{RFC 2845 <https://tools.ietf.org/html/rfc2845>}.
+
+    @type algorithm: L{Name}
+    @ivar algorithm: The name of the signature or MAC algorithm.
+
+    @type time: L{int}
+    @ivar time: Signing time, as seconds from the POSIX epoch.
+
+    @type fudge: L{int}
+    @ivar fudge: Allowable time skew, in seconds.
+
+    @type MAC: L{bytes}
+    @ivar MAC: The message digest or signature.
+
+    @type originalID: L{int}
+    @ivar originalID: A message ID.
+
+    @type otherData: L{bytes}
+    @ivar otherData: Other data carried in exceptional cases.
+
+    """
+    fancybasename = "TSIG"
+    compareAttributes = ('algorithm', 'time', 'fudge',
+                         'MAC', 'originalID', 'error', 'otherData',
+                         'ttl')
+    showAttributes = ['algorithm', 'time', 'MAC', 'error', 'otherData']
+
+    TYPE = TSIG
+
+    def __init__(self, algorithm=None, time=None,
+                 fudge=5, MAC=None, originalID=0,
+                 error=OK, otherData=b'', ttl=0):
+        self.algorithm = None if algorithm is None else Name(algorithm)
+        self.time = None if time is None else int(time)
```

Comment:
we do we need this?
Each if/else branch need a test... so I think that we should just assing `self.timeSigned = timeSigned` and don't try to add any more logic.

Otherwise, please document and **test** that it also accepts text/str for time.
I don't see any test where time is None or not an int.